src/main/java/org/mockito/internal/junit/JUnitRule.java

```diff
@@ -6,74 +6,29 @@
 
 import org.junit.runners.model.FrameworkMethod;
 import org.junit.runners.model.Statement;
-import org.mockito.Mockito;
-import org.mockito.MockitoAnnotations;
-import org.mockito.MockitoSession;
-import org.mockito.internal.session.MockitoSessionLoggerAdapter;
 import org.mockito.junit.MockitoRule;
 import org.mockito.plugins.MockitoLogger;
 import org.mockito.quality.Strictness;
 
-/**
- * Internal implementation.
- */
-public class JUnitRule implements MockitoRule {
-
-    private final MockitoLogger logger;
-    private Strictness strictness;
-    private MockitoSession session;
+/** Internal implementation. */
+public final class JUnitRule extends AbstractJUnitRule implements MockitoRule {
```

Comment:
Abstract class hierarchies are complex. Is there a way to share the code without the abstract class? Delegation or composition? Thoughts?

Reply:
Done